examples/AvroGeneric/Program.cs

```diff
@@ -0,0 +1,121 @@
+                return;
+            }
+
+            string bootstrapServers = args[0];
+            string schemaRegistryUrl = args[1];
+            string topicName = args[2];
+
+            var producerConfig = new Dictionary<string, object>
+            {
+                { "bootstrap.servers", bootstrapServers },
+                // Note: you can specify more than one schema registry url using the
+                // schema.registry.url property for redundancy (comma separated list).
+                // The property name is not plural to follow the convention set by
+                // the Java implementation.
+                { "schema.registry.url", schemaRegistryUrl },
+                // optional schema registry client properties:
+                { "schema.registry.connection.timeout.ms", 5000 },
+                { "schema.registry.max.cached.schemas", 10 },
+                // optional avro serializer properties:
+                { "avro.serializer.buffer.bytes", 50 },
+                { "avro.serializer.auto.register.schemas", true }
+            };
+
+            var consumerConfig = new Dictionary<string, object>
+            {
+                { "bootstrap.servers", bootstrapServers },
+                { "group.id", Guid.NewGuid() },
+                { "schema.registry.url", schemaRegistryUrl }
+            };
+
+            var s = (RecordSchema)Schema.Parse(
+                @"{
+                    ""namespace"": ""Confluent.Kafka.Examples.AvroSpecific"",
+                    ""type"": ""record"",
+                    ""name"": ""User"",
+                    ""fields"": [
+                        {""name"": ""name"", ""type"": ""string""},
+                        {""name"": ""favorite_number"",  ""type"": [""int"", ""null""]},
+                        {""name"": ""favorite_color"", ""type"": [""string"", ""null""]}
+                    ]
+                  }"
+            );
+            
+            using (var consumer = new Consumer<string, GenericRecord>(consumerConfig, new AvroDeserializer<string>(), new AvroDeserializer<GenericRecord>()))
+            using (var producer = new Producer<string, GenericRecord>(producerConfig, new AvroSerializer<string>(), new AvroSerializer<GenericRecord>()))
+            {
+                consumer.OnMessage += (o, e)
+                    => Console.WriteLine($"Key: {e.Key}\nValue: {e.Value}");
+
+                consumer.OnError += (_, e)
+                    => Console.WriteLine("Error: " + e.Reason);
+
+                consumer.OnConsumeError += (_, e)
+                    => Console.WriteLine("Consume error: " + e.Error.Reason);
+
+                consumer.Subscribe(topicName);
+
+                CancellationTokenSource cts = new CancellationTokenSource();
+                var consumeTask = Task.Factory.StartNew(() =>
+                {
+                    while (!cts.Token.IsCancellationRequested)
```

Comment:
Rumour on the street is that this should be in a try-block due to AggregationExceptions ;)

Reply:
haha